library/core/tests/array.rs

```diff
@@ -330,3 +330,25 @@ fn array_map_drop_safety() {
     assert_eq!(DROPPED.load(Ordering::SeqCst), num_to_create);
     panic!("test succeeded")
 }
+
+#[test]
+fn build_array() {
+    let array: [usize; 5] = core::array::build_array(|i| i);
+    assert_eq!(array, [0, 1, 2, 3, 4]);
+}
+
+#[test]
+fn try_build_array() {
+    use core::array::try_build_array;
+
+    #[derive(Debug, PartialEq)]
+    enum SomeError {
+        Foo,
+    }
+
+    let array: Result<[usize; 5], SomeError> = try_build_array(|i| Ok(i));
+    assert_eq!(array, Ok([0, 1, 2, 3, 4]));
+
+    let another_array: Result<[usize; 5], SomeError> = try_build_array(|_| Err(SomeError::Foo));
+    assert_eq!(another_array, Err(SomeError::Foo));
+}
```

Comment:
I think that a drop test might also be useful to demonstrate the expected behaviour of frees when it errors.

Reply:
Indeed! Thanks for pointing out this scenario

Reply:
Now that code is unified, proper handling on panics should already be covered by the `array_map_drop_safety` test that you created but I added another similar test "just in case".